projects/com.oracle.truffle.llvm.tools/src/com/oracle/truffle/llvm/tools/Clang.java

```diff
@@ -79,7 +79,7 @@ public static void main(String[] args) {
     public static void compileToLLVMIR(File path, File destinationFile, ClangOptions options) {
         File tool = getCompileToolFromExtension(path);
         String[] command = new String[]{tool.getAbsolutePath(), "-I " + LLVMBaseOptionFacade.getProjectRoot() + "/../include", emitLLVMIRTo(destinationFile), optimizationLevel(options),
-                        path.getAbsolutePath()};
+                        "-c ", path.getAbsolutePath()};
```

Comment:
Seems without -c option it is  going to linking phase and fails  with linker error and it fails to generate .bc file.

Executing compilation command explicit compilation showed the following,
.....projects/com.oracle.truffle.llvm.tools/tools/llvm/bin/../lib/LLVMgold.so: error loading plugin
.....projects/com.oracle.truffle.llvm.tools/tools/llvm/bin/../lib/LLVMgold.so: error in plugin cleanup (ignored)

Would you like me to investigate further?